controllers/complianceeventsapi/server.go

```diff
@@ -0,0 +1,334 @@
+		WriteTimeout: 15 * time.Second,
+		IdleTimeout:  15 * time.Second,
+	}
+
+	db, err := sql.Open("postgres", dbURL)
+	if err != nil {
+		return err
+	}
+
+	goquDB := goqu.New("postgres", db)
+
+	// register handlers here
+	mux.HandleFunc("/api/v1/compliance-events", func(w http.ResponseWriter, r *http.Request) {
+		switch r.Method {
+		case http.MethodPost:
+			postComplianceEvent(goquDB)(w, r)
+		default:
+			http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+		}
+	})
+
+	go func() {
+		err := s.server.ListenAndServe()
+		if err != nil && err != http.ErrServerClosed {
+			log.Error(err, "Error starting compliance events api server")
+		}
+	}()
+
+	s.isRunning = true
+
+	return nil
+}
+
+// Stop stops the http server. If it is not currently running, it has no effect.
+func (s *databaseAPIServer) Stop() {
+	s.Lock.Lock()
+	defer s.Lock.Unlock()
+
+	if !s.isRunning {
+		return
+	}
+
+	if err := s.server.Shutdown(context.TODO()); err != nil {
+		log.Error(err, "Error stopping compliance events api server")
+	}
+
+	s.isRunning = false
+}
+
+func postComplianceEvent(goquDB *goqu.Database) func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		body, err := io.ReadAll(r.Body)
+		if err != nil {
+			log.Error(err, "error reading")
+			http.Error(w, "Error reading request body", http.StatusBadRequest)
+
+			return
+		}
+
+		reqEvent := &ComplianceEvent{}
+
+		if err := json.Unmarshal(body, reqEvent); err != nil {
+			log.Error(err, "error unmarshalling")
+			http.Error(w, "Incorrectly formatted request body", http.StatusBadRequest)
+
+			return
+		}
+
+		if err := reqEvent.Validate(); err != nil {
+			log.Error(err, "error validating")
+			http.Error(w, "Invalid request body", http.StatusBadRequest)
+
+			return
+		}
+
+		clusterFK, err := getClusterForeignKey(r.Context(), goquDB, reqEvent.Cluster)
+		if err != nil {
+			log.Error(err, "error getting cluster foreign key")
+			http.Error(w, "Internal Error", http.StatusInternalServerError)
+
+			return
+		}
+
+		reqEvent.Event.ClusterID = clusterFK
+
+		if reqEvent.ParentPolicy != nil {
+			pfk, err := getParentPolicyForeignKey(r.Context(), goquDB, *reqEvent.ParentPolicy)
+			if err != nil {
+				log.Error(err, "error getting parent policy foreign key")
+				http.Error(w, "Internal Error", http.StatusInternalServerError)
+
+				return
+			}
+
+			reqEvent.Policy.ParentPolicyID = &pfk
+		}
+
+		policyFK, err := getPolicyForeignKey(r.Context(), goquDB, reqEvent.Policy)
+		if err != nil {
+			log.Error(err, "error getting policy foreign key")
+			http.Error(w, "Internal Error", http.StatusInternalServerError)
+
+			return
+		}
+
+		reqEvent.Event.PolicyID = policyFK
+
+		insert := goquDB.Insert("compliance_events").Rows(reqEvent.Event).Executor()
+
+		_, err = insert.Exec()
+		if err != nil {
+			log.Error(err, "error inserting compliance event")
+			http.Error(w, "Internal Error", http.StatusInternalServerError)
+		}
```

Comment:
On success, the status code should be 201 and the returned JSON should likely be the serialized form of `reqEvent`. I can also handle the return value in my task on the GET HTTP endpoint.

Reply:
It should be easy enough for me to put it in here.